File: zig/space-age/space_age.zig
pub const Planet = enum {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    pub fn age(self: Planet, seconds: usize) f64 {
        const earthYears = @as(f64, @floatFromInt(seconds)) / 31557600;
        switch (self) {
            .mercury => return earthYears / 0.2408467,
            .venus => return earthYears / 0.61519726,
            .earth => return earthYears,
            .mars => return earthYears / 1.8808158,
            .jupiter => return earthYears / 11.862615,
            .saturn => return earthYears / 29.447498,
            .uranus => return earthYears / 84.016846,
            .neptune => return earthYears / 164.79132,
        }
    }
};

File: solutions/php/space-age/1/SpaceAge.php
<?php
declare(strict_types=1);

class SpaceAge
{
    private float $age;

    function __construct(int $seconds)
    {
        $this->age = $seconds / 31557600;
    }

    function earth(): float
    {
        return $this->age;
    }

    function mercury(): float
    {
        return $this->age / 0.2408467;
    }

    function venus(): float
    {
        return $this->age / 0.61519726;
    }

    function mars(): float
    {
        return $this->age / 1.8808158;
    }

    function jupiter(): float
    {
        return $this->age / 11.862615;
    }

    function saturn(): float
    {
        return $this->age / 29.447498;
    }

    function uranus(): float
    {
        return $this->age / 84.016846;
    }

    function neptune(): float
    {
        return $this->age / 164.79132;
    }
}
